keg_storage/backends/azure.py

```diff
@@ -0,0 +1,247 @@
+
+    def _get_blob_size(self):
+        """
+        Retrieve the overall size of the blob
+        """
+        blob = self.service.get_blob_properties(
+            container_name=self.container_name,
+            blob_name=self.path
+        )
+        return blob.properties.content_length
+
+    def _refill_buffer(self):
+        """
+        Retrieve the next chunk and add it to the local buffer.
+        """
+        range_end = self.current_offset + self.chunk_size - 1  # -1 because range is inclusive
+        blob = self.service.get_blob_to_bytes(
+            container_name=self.container_name,
+            blob_name=self.path,
+            start_range=self.current_offset,
+            end_range=range_end,
+        )
+        self.current_offset += len(blob.content)
+        self.buffer.extend(blob.content)
+
+    def _read_from_buffer(self, max_size):
+        """
+        Read up to max_size bytes from the local buffer.
+        """
+        read_size = min(len(self.buffer), max_size)
+        output = self.buffer[:read_size]
+        self.buffer = self.buffer[read_size:]
+        self.read_position += len(output)
+        return output
+
+    def read(self, size: int) -> bytes:
+        output_buf = bytes()
+
+        # Read up to the end of the file or `size` bytes whichever is less
+        read_size = min(self.blob_size - self.read_position, size)
+
+        while len(output_buf) < read_size:
+            if len(self.buffer) == 0:
+                # Buffer is empty, refill it.
+                self._refill_buffer()
+
+            remaining = read_size - len(output_buf)
+            read = self._read_from_buffer(remaining)
+            output_buf += read
+
+        return output_buf
+
+
+class AzureStorage(base.StorageBackend):
+    def __init__(self, account: str, key: str, bucket: str, name: str = 'azure'):
+        super().__init__()
+        self.name = name
+        self.account = account
+        self.key = key
+        self.bucket = bucket
+
+    def _create_service(self):
+        return BlockBlobService(
+            account_name=self.account,
+            account_key=self.key,
+        )
+
+    def list(self, path: str) -> typing.List[base.ListEntry]:
+        service = self._create_service()
+        if not path.endswith('/'):
+            path = path + '/'
+        list_iter = service.list_blobs(container_name=self.bucket, prefix=path, delimiter='/')
+
+        def construct_entry(blob):
+            if isinstance(blob, BlobPrefix):
+                return base.ListEntry(name=blob.name, last_modified=None, size=0)
+            return base.ListEntry(
+                name=blob.name,
+                last_modified=arrow.get(blob.properties.last_modified),
+                size=blob.properties.content_length,
+            )
+
+        return [construct_entry(blob) for blob in list_iter]
+
+    def open(self, path: str, mode: base.FileMode, buffer_size: int = 10 * 1024 * 1024) -> AzureFile:  # noqa
```

Comment:
Can we add a helper method that allows a `str` or `base.FileMode` and will convert the string to a `FileMode`?